src/tools/node_maintain.py

```diff
@@ -236,6 +114,211 @@ def convert_nodes(nodes_str):
     return set()
 
 
+def validate_vc_name(vc_name_str):
+    if re.match(r"^[A-Za-z0-9_]+$", vc_name_str) is None:
+        raise argparse.ArgumentTypeError("invalid vc name: {}. Only alphanumeric and _ allowed".format(vc_name_str))
+    return vc_name_str
+
+
+def is_dedicated_vc(queue_name, queue_attr):
+    # print(json.dumps(queue_attr, indent=2))
+    if queue_name == "" or queue_name == "*" or queue_attr["defaultNodeLabelExpression"] != queue_name:
+        return False
+    if queue_name not in queue_attr["capacities"] or queue_attr["capacities"][queue_name]["maxCapacity"] != 100:
+        return False
+    return True
+
+
+def get_resource_by_label(nodes_info):
+    labels_dict = {}
+    default_resource = Resource(**{"cpus": 0, "memory": 0, "gpus": 0})
+    for node, info in nodes_info.iteritems():
+        if info["nodeLabel"] not in labels_dict:
+            labels_dict[info["nodeLabel"]] = {
+                "resource": default_resource
+            }
+        labels_dict[info["nodeLabel"]]["resource"] += info["resource"]
+    return labels_dict
+
+
+def get_dedicate_vc(args):
+    yarn_operator = YarnOperator(args.resource_manager_ip)
+    queues_info = yarn_operator.get_queues_info()
+    nodes_info = yarn_operator.get_nodes_info()
+    dedicate_queues = {queue_name: {"resource": Resource(**{"cpus": 0, "memory": 0, "gpus": 0}), "nodes": []} for queue_name, queue_info in queues_info.items() if
+                       is_dedicated_vc(queue_name, queue_info)}
+    if len(dedicate_queues) == 0:
+        logger.info("No dedicated vc found")
+        return
+
+    labeled_resources = get_resource_by_label(nodes_info)
+    for partition in labeled_resources:
+        if partition in dedicate_queues:
+            dedicate_queues[partition]["resource"] = labeled_resources[partition]["resource"]
+
+    for node in nodes_info:
+        if nodes_info[node]["nodeLabel"] in dedicate_queues:
+            dedicate_queues[nodes_info[node]["nodeLabel"]]["nodes"].append(node)
+    for queue_name, queue_attr in dedicate_queues.items():
+        print(queue_name + ":")
```

Comment:
is this file py2 or py3? could you make it compatible?

Reply:
it's a dev-box script with python2, I made some compatible fix, but not fully tested, expecially unittest part.